be/src/runtime/statistic_result_writer.cpp

```diff
@@ -60,13 +60,13 @@ Status StatisticResultWriter::append_chunk(vectorized::Chunk* chunk) {
 
     int version = down_cast<Int32Column*>(ColumnHelper::get_data_column(result_columns[0].get()))->get_data()[0];
 
-    auto* result = new (std::nothrow) TFetchDataResult();
+    std::unique_ptr<TFetchDataResult> result(new (std::nothrow) TFetchDataResult());
```

Comment:
Do the following code handle the `std::nothrow` strategy

Reply:
> Do the following code handle the `std::nothrow` strategy

I'm afraid it doesn't, so I add a check before use it